gtest_parallel_tests.py

```diff
@@ -300,5 +303,133 @@ def test_get_save_file_path_win32(self):
                          gtest_parallel.get_save_file_path())
 
 
+class TestSerializeTestCases(unittest.TestCase):
+  class TaskManagerMock(object):
+    def __init__(self, group_checker):
+      self.group_checker = group_checker
+      self.check_lock = threading.Lock()
+
+    def run_task(self, task):
+      test_group = task.test_name.split('.')[0]
+
+      with self.check_lock:
+        self.group_checker.add(test_group)
+
+      # Delay as if real test were run.
+      time.sleep(0.001 * task.execution_number)
```

Comment:
I realized that I got meaning of task.execution_number wrong. However, even without this code tests should be fine.